macros/remove_string_from_numeric.sql

```diff
@@ -0,0 +1,11 @@
+{% macro remove_string_from_numeric(column_name) %}
+
+{% if target.type == 'bigquery' %}
+    cast(regexp_replace(cast({{ column_name }} as {{ dbt.type_string() }}), r'[^0-9.]*', '') as {{ dbt.type_numeric() }})
+{% elif target.type == 'postgres' %}
+    cast(regexp_replace(cast({{ column_name }} as {{ dbt.type_string() }}), '[^0-9.]*', '', 'g') as {{ dbt.type_numeric() }})
+{% else %}
+    cast(regexp_replace(cast({{ column_name }} as {{ dbt.type_string() }}), '[^0-9.]*', '') as {{ dbt.type_numeric() }})
+{% endif %}
+
+{% endmacro %}
```

Comment:
@fivetran-avinash this is the right idea. However, when creating a macro that is cross warehouse compatibile we don't want to leverage the hard `target.type` conditional coding logic. Especially if there are plans to one day have this integrated into fivetran_utils. Instead, we want to leverage the `adapter.dispatch` to handle these operations for us. 

Would you be able to rework this macro to follow the `adapter.dispatch` framework of dispatching, as opposed to using the `target.type` conditionals? You can find an example of an adapter.dispatch framework macro [here](https://github.com/fivetran/dbt_fivetran_utils/blob/v0.4.10/macros/json_parse.sql). Your macro will look similar, but will follow the new macro format for each warehouse. However, you will be able to leverage the `default__` macro as the same as your `else` version. Let me know if you have any questions.